packages/gatsby-theme-docs/src/components/multi-language-code-examples.js

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useReducer } from 'react';
+import PropTypes from 'prop-types';
+import { ContentNotifications, CodeBlock } from '@commercetools-docs/ui-kit';
+import useCodeExamples from '../hooks/use-code-examples';
+
+function reducer(state, action) {
+  switch (action.type) {
+    case 'set_props':
+      return { props: action.props, ...state };
+    case 'set_errors':
+      return { errors: action.errors, ...state };
+    default:
+      throw new Error(`"${action.type}" is invalid.`);
+  }
+}
+
+function MultiLanguageCodeExamples(props) {
+  const codeExamples = useCodeExamples();
+  const [state, dispatch] = useReducer(reducer, {});
+
+  useEffect(() => {
+    extractProps({
+      children: props.children,
+      codeExamples,
+      callback: (result, err) => {
+        if (result) {
+          dispatch({ type: 'set_props', props: result });
+        } else {
+          dispatch({ type: 'set_errors', errors: err });
+        }
+      },
+    });
+  }, [codeExamples, props.children]);
+
+  if (state.errors) {
+    if (process.env.NODE_ENV !== 'production') {
+      return state.errors.map((err, index) => (
+        <ContentNotifications.Error key={index}>
+          {err}
+        </ContentNotifications.Error>
+      ));
+    }
+
+    throw new Error(state.errors);
+  }
+
+  return state.props ? (
+    <CodeBlock multiLanguage={{ title: props.title, props: state.props }} />
+  ) : null;
+}
+
+function extractProps({ children, codeExamples, callback }) {
+  const errors = [];
+  const props = [];
+
+  children.forEach(child => {
+    if (!child.props) {
+      errors.push(
+        `Content of MultiLanguageCodeExamples must be a CodeExample and not ${JSON.stringify(
+          child
+        )}`
+      );
+    } else if (!(child.props.mdxType === 'CodeExample')) {
+      errors.push(
+        `Content of MultiLanguageCodeExamples must be a CodeExample and not "${child.props.mdxType}"`
+      );
+    } else {
+      const codeExample = codeExamples.find(example => {
+        return example.absolutePath.includes(child.props.file);
+      });
+
+      if (!codeExample) {
+        errors.push(`Code example does not exist for ${child.props.file}`);
+      } else {
+        const language = child.props.file.split('.').pop();
+
+        props.push({
+          content: codeExample.content,
+          language,
+          highlightLines: child.props.highlightLines || [],
+          noPromptLines: child.props.noPromptLines || [],
+        });
+      }
+    }
+  });
+
+  if (errors.length) {
+    callback(undefined, errors.length ? errors : '');
+  } else {
+    callback(props);
+  }
+}
+
+MultiLanguageCodeExamples.propTypes = {
+  title: PropTypes.string,
+  children: PropTypes.arrayOf(PropTypes.element),
```

Comment:
Shouldn't the props be required?

Reply:
If you mean props of a child, there is no need to make it required since an error message will be rendered to authors - props needs to be validated within the component to do that.

Reply:
I mean this:

```suggestion
  children: PropTypes.arrayOf(PropTypes.element.isRequired).isRequired,
```

It does not have anything to do with the authors and the custom validation, it's simply from a static type perspective.